datalad/distribution/tests/test_dataset.py

```diff
@@ -296,57 +298,51 @@ def test_require_dataset():
 
 @with_tempfile(mkdir=True)
 def test_dataset_id(path):
+
+    import gc
+
     ds = Dataset(path)
     assert_equal(ds.id, None)
     ds.create()
     dsorigid = ds.id
     # ID is always a UUID
     assert_equal(ds.id.count('-'), 4)
     assert_equal(len(ds.id), 36)
+
+    # Ben: The following part of the test is concerned with creating new objects
+    #      and therefore used to reset the flyweight dict while keeping a ref to
+    #      the old object for comparison etc. This is ugly and in parts
+    #      retesting what is already tested in `test_Dataset_flyweight`. No need
+    #      for that. If we del the last ref to an instance and gc.collect(),
+    #      then we get a new instance on next request. This test should trust
+    #      the result of `test_Dataset_flyweight`.
```

Comment:
Should we strip the test down to cover only what is not tested in `test_Dataset_flyweight` then?

Reply:
I think that's what I did. No assertions about the objects anymore. Just dataset id.